sycl/doc/extensions/proposed/SYCL_EXT_ONEAPI_GROUP_STATIC_ARRAY.asciidoc

```diff
@@ -0,0 +1,590 @@
+Default layout is one that is specified by the `close` property.
+`in_data` mustn't be changed by the algorithm.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+  template <typename Group, typename T, std::size_t ItemsPerWorkItem, typename BinaryOperation, typename... Properties>
+  void inclusive_scan_over_group(Group g, sycl::span<T, ItemsPerWorkItem> in_data, sycl::span<T, ItemsPerWorkItem> out_data, BinaryOperation binary_op, sycl::ext::oneapi::experimental::properties properties = {}); // (1)
+
+  template <typename Group, typename V, std::size_t ItemsPerWorkItem, typename T, typename BinaryOperation, typename... Properties>
+  void inclusive_scan_over_group(Group g, sycl::span<V, ItemsPerWorkItem> in_data, sycl::span<V, ItemsPerWorkItem> out_data, BinaryOperation binary_op, T init, sycl::ext::oneapi::experimental::properties properties = {}); // (2)
+
+}
+----
+
+_Preconditions_: `binary_op` must be an instance of a SYCL function object.
+
+1._Constraints_: Available only if `sycl::is_group_v<std::decay_t<Group>>` is true,
+`T` is a fundamental type, `BinaryOperation` is a SYCL function object type,
+`ItemsPerWorkItem` does not equal to `sycl::dynamic_extent`.
+
+_Mandates_: `binary_op(in_data[i], in_data[j])`, where `i, j < ItemsPerWorkItem`,
+must return a value of type `T`.
+
+_Effects_: Perform the inclusive scan of all values of `in_data` for all work-items in the group
+and the identity value of `binary_op`, using the operator `binary_op`. The scan is computed
+using a generalized noncommutative sum as defined in standard C++. For multi-dimensional groups,
+the order of work-items in the group is determined by their linear id.
+Result is placed into `out_data` with the layout specified by `properties`.
+Default layout is one that is specified by the `close` property.
+`in_data` mustn't be changed by the algorithm.
+
+2._Constraints_: Available only if `sycl::is_group_v<std::decay_t<Group>>` is true,
+`V` and `T` are fundamental types, `BinaryOperation` is a SYCL function object type and
+`ItemsPerWorkItem` does not equal to `sycl::dynamic_extent`.
+
+_Mandates_: `binary_op(init, in_data[i])`, where `i < ItemsPerWorkItem`,
+must return a value of type `T`.
+
+_Effects_: Perform the inclusive scan of all values of `in_data` for all work-items in the group
+and an initial value specified by `init`, using the operator `binary_op`. The scan is computed
+using a generalized noncommutative sum as defined in standard C++. For multi-dimensional groups,
+the order of work-items in the group is determined by their linear id.
+Result is placed into `out_data` with the layout specified by `properties`.
+Default layout is one that is specified by the `close` property.
+`in_data` mustn't be changed by the algorithm.
+
+=== Sorting functions
+
+Sorting function is a SYCL Sorting Extension.
+
+Following functions perform sorting including key-value variant.
```

Comment:
I think it would make more sense to move these extensions to the sorting APIs into the SYCL_EXT_ONEAPI_GROUP_SORT extension.  That way, a user will see all of the sorting APIs in one place.

If you wanted, you could define a new "version" (version 2) of the SYCL_EXT_ONEAPI_GROUP_SORT extension, and these new APIs would only be available in version 2.  This would allow an application to test the feature-test macro in order to know whether the new APIs are supported.  However, since SYCL_EXT_ONEAPI_GROUP_SORT is an experimental API, we don't have strong compatibility requirements, so it would be OK to add the new APIs without changing the feature-test macro version.

Note that this extension is "proposed" while the existing SYCL_EXT_ONEAPI_GROUP_SORT extension is "experimental" (i.e. already implemented).  You can create a proposed revision to an existing extension by doing:

* Copy the existing "sycl/doc/experimental/SYCL_EXT_ONEAPI_GROUP_SORT.asciidoc" into "sycl/doc/proposed".
* Make the proposed API changes to the version in "sycl/doc/proposed".
* When the proposed APIs are implemented, we will move "sycl/doc/proposed/SYCL_EXT_ONEAPI_GROUP_SORT.asciidoc" to "sycl/doc/experimental", overwriting the previous version.

Reply:
@gmlueck, thanks for your feedback. It seems applicable.
Let me think how it will look like.